azure_monitor/src/azure_monitor/sdk/auto_collection/live_metrics/dependency_metrics.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import time
+from typing import Dict
+
+from opentelemetry.metrics import Meter
+
+from azure_monitor.sdk.auto_collection.metrics_span_processor import (
+    AzureMetricsSpanProcessor,
+)
+
+dependency_map = dict()
+
+
+class DependencyLiveMetrics:
+    """Starts auto collection of dependency live metrics.
+
+    Args:
+        meter: OpenTelemetry Meter
+        labels: Dictionary of labels
+    """
+
+    def __init__(
+        self,
+        meter: Meter,
+        labels: Dict[str, str],
+        span_processor: AzureMetricsSpanProcessor,
+    ):
+        self._meter = meter
+        self._labels = labels
+        self._span_processor = span_processor
+
+        meter.register_observer(
+            callback=self._track_dependency_duration,
+            name="\\ApplicationInsights\\Dependency Call Duration",
+            description="Average Outgoing Requests duration",
+            unit="milliseconds",
+            value_type=float,
+        )
+        meter.register_observer(
+            callback=self._track_failure_rate,
+            name="\\ApplicationInsights\\Dependency Calls Failed\/Sec",
+            description="Failed Outgoing Requests per second",
+            unit="rps",
+            value_type=int,
+        )
+
+    def _track_dependency_duration(self, observer) -> None:
+        """ Track Dependency average duration
+
+        Calculated by getting the time it takes to make an outgoing request
+        and dividing over the amount of outgoing requests over an elapsed time.
+        """
+        last_average_duration = dependency_map.get("last_average_duration", 0)
+        interval_duration = (
+            self._span_processor.dependency_duration
+            - dependency_map.get("last_duration", 0)
+        )
+        interval_count = (
```

Comment:
Is there someway to abstract the logic between livemetrics and autocollection? Or is this not possible because they have separate export intervals?